### DISCUSSION TITLE
Dark mode using Livewire v3 and tailwind

### DISCUSSION THREAD

**Discussion by aminulloh2002** [livewire/livewire#6832] [1. Help]
so i want to use dark mode using tailwind,

here is my code 
```
 <body class="flex font-sans antialiased" x-data="switchTheme()" :class="{ 'dark': darkTheme}">
        @include('layouts.sidebar')
    .....

        <script>
            function switchTheme() {
                return {
                    darkTheme: localStorage.getItem('darkTheme') == 'true',
                    toggleTheme() {
                        this.darkTheme = !this.darkTheme;
                        localStorage.setItem('darkTheme', this.darkTheme);
                    }
                }
            }
        </script>
</body>
```

when i try to navigate using <a href="..." wire:navigate>

the dark mode will "blink", the bod will be rerendered and waiting the alpine js to add the dark class to the body.

any help will be appreciated, thanks in advance.

**Comment by LukasJK** [comment]
Curious here as well if there's some clever way to preserve alpine state

**Reply by WilliamPinto-Olmos** [reply]
@LukasJK I think i saw a comment saying that if you use the `:class="{ 'dark': darkTheme}"` on the HTML instead of the Body tag it wont "blink", since Livewire navigate only replaces the body tag. 

And for persisting alpine state i think you can use the [persist plugin](https://alpinejs.dev/plugins/persist) which is included by default in the version of alpinejs bundled with LivewireV3. Not sure if the plugin is what you need tho
